src/bp-messages/classes/class-bp-messages-thread.php

```diff
@@ -1458,4 +1458,170 @@ public static function update_tables() {
 
 		return (bool) ! $errors;
 	}
+
+	public static function get( $args ) {
+		global $wpdb;
+
+		$bp = buddypress();
+
+		$defaults = array(
+			'orderby'  => 'id',
+			'order'    => 'DESC',
+			'per_page' => 20,
+			'page'     => 1,
+			'user_id'  => 0,
+			'include'  => false,
+			'exclude'  => false,
+		);
+
+		$r = bp_parse_args( $args, $defaults, 'bp_recipients_recipient_get' );
+
+		$sql = array(
+			'select'     => 'SELECT DISTINCT r.id',
+			'from'       => "{$bp->messages->table_name_recipients} r",
+			'where'      => '',
+			'orderby'    => '',
+			'pagination' => '',
+		);
+
+		$where_conditions = array();
+
+		if ( ! empty( $r['include'] ) ) {
+			$include                     = implode( ',', wp_parse_id_list( $r['include'] ) );
+			$where_conditions['include'] = "r.id IN ({$include})";
+		}
+
+		if ( ! empty( $r['exclude'] ) ) {
+			$exclude                     = implode( ',', wp_parse_id_list( $r['exclude'] ) );
+			$where_conditions['exclude'] = "r.id NOT IN ({$exclude})";
+		}
+
+		/* Order/orderby ********************************************/
+
+		$order   = $r['order'];
+		$orderby = $r['orderby'];
+
+		// Sanitize 'order'.
+		$order = bp_esc_sql_order( $order );
+
+		/**
+		 * Filters the converted 'orderby' term.
+		 *
+		 * @since BuddyBoss 1.5.4
+		 *
+		 * @param string $value   Converted 'orderby' term.
+		 * @param string $orderby Original orderby value.
+		 * @param string $value   Parsed 'type' value for the get method.
```

Comment:
@janakkaneriya007 Removed this param doc as param does not exit also check on Message nad notice docs